patterns/design-patterns-2/src/main/java/com/baeldung/prototype/ShapeCache.java

```diff
@@ -0,0 +1,19 @@
+package com.baeldung.prototype;
+
+import java.util.Map;
+
+public class ShapeCache {
+
+    private final Map<String,Shape> shapeCache;
```

Comment:
The `final` ness of this doesn't add much. If you're trying to make things immutable, this wouldn't do it, and this is not an immutable class.